lib/boombox/bin.ex

```diff
@@ -0,0 +1,255 @@
+defmodule Boombox.Bin do
+  @moduledoc """
+  Boombox.Bin is a Membrane Bin for audio and video streaming.
+  It can be used as a Sink or Source in your Membrane Pipeline.
+
+  If you use it as a Membrane Source, you have to specify `:input` option.
+  If you use it as a Membrane Sink, you have to specify `:output` option.
+  Boombox.Bin cannot be used as both Source and Sink at the same time.
```

Comment:
Maybe mention what pads they need to link too, and provide some example, like `child(%Boombox.Bin{input: "file.mp4"}) |> via_out(:output, options: [kind: :video]) |> ...`